internal/impl/shell.go

```diff
@@ -64,14 +64,15 @@ type DevboxShell struct {
 
 	// shellStartTime is the unix timestamp for when the command was invoked
 	shellStartTime string
+	pure           bool
 }
 
 type ShellOption func(*DevboxShell)
 
 // NewDevboxShell initializes the DevboxShell struct so it can be used to start a shell environment
 // for the devbox project.
-func NewDevboxShell(nixpkgsCommitHash string, opts ...ShellOption) (*DevboxShell, error) {
-	shPath, err := shellPath(nixpkgsCommitHash)
+func NewDevboxShell(nixpkgsCommitHash string, pure bool, opts ...ShellOption) (*DevboxShell, error) {
```

Comment:
Make `NewDevboxShell` and `shellPath` have devbox as receiver that way you don't need to add `pure` param.